akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala

```diff
@@ -587,8 +587,7 @@ private[akka] class ClusterShardingGuardian extends Actor {
 
     case StartProxy(typeName, dataCenter, settings, extractEntityId, extractShardId) ⇒
       try {
-
-        val encName = URLEncoder.encode(typeName, ByteString.UTF_8)
+        val encName = URLEncoder.encode(s"${typeName}Proxy", ByteString.UTF_8)
```

Comment:
It should also be changed in the `ClusterSharding` extension. Two places there

* `def proxyName`: https://github.com/akka/akka/blob/master/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala#L386
* `def shardRegion`: https://github.com/akka/akka/blob/master/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala#L453

The latter should try both names, so something like this:
```
  def shardRegion(typeName: String): ActorRef = regions.get(typeName) match {
    case null ⇒ regions.get(proxyName(typeName, None)) match {
      case null => throw new IllegalArgumentException(s"Shard type [$typeName] must be started first")
      case ref => ref
    }
    case ref  ⇒ ref
  }
```